e2e/sample-apps/compat.js

```diff
@@ -0,0 +1,270 @@
+ * Create, read, delete.
+ */
+async function callStorage() {
+  console.log('[STORAGE] start');
+  const storage = firebase.storage();
+  const storageRef = storage.ref('/test.txt');
+  await storageRef.putString('efg');
+  await new Promise(resolve => setTimeout(resolve, 1000));
+  const url = await storageRef.getDownloadURL();
+  console.log('[STORAGE] download url', url);
+  const response = await fetch(url);
+  const data = await response.text();
+  console.log("[STORAGE] Returned data (should be 'efg'):", data);
+  await storageRef.delete();
+}
+
+/**
+ * Firestore smoke test.
+ * Create 2 docs, test query filter.
+ * Create, update, delete a doc with `onSnapshot` monitoring changes.
+ */
+async function callFirestore() {
+  console.log('[FIRESTORE] start');
+  const firestore = firebase.firestore();
+  await firestore.collection('testCollection').doc('trueDoc').set({
+    testbool: true
+  });
+  await firestore.collection('testCollection').doc('falseDoc').set({
+    testbool: false
+  });
+  const trueDocs = await firestore
+    .collection('testCollection')
+    .where('testbool', '==', true)
+    .get();
+  trueDocs.docs.forEach(doc =>
+    console.log('[FIRESTORE] Filter test, expect one doc', doc.data())
+  );
+  await firestore.collection('testCollection').doc('trueDoc').delete();
+  await firestore.collection('testCollection').doc('falseDoc').delete();
+  const testDocRef = firestore.doc('testCollection/testDoc');
+  console.log('[FIRESTORE] Doc creation and updating');
+  testDocRef.onSnapshot(snap => {
+    if (snap.exists) {
+      console.log('[FIRESTORE] SNAPSHOT:', snap.data());
+    } else {
+      console.log("[FIRESTORE] Snapshot doesn't exist");
+    }
+  });
+  console.log('[FIRESTORE] creating (expect to see snapshot data)');
+  await testDocRef.set({ word: 'hi', number: 14 });
+  console.log('[FIRESTORE] updating (expect to see snapshot data change)');
+  await testDocRef.update({ word: 'bye', newProp: ['a'] });
+  console.log("[FIRESTORE] deleting (expect to see snapshot doesn't exist)");
+  await testDocRef.delete();
+}
+
+/**
+ * Database smoke test.
+ * Create, update, delete a doc with `on` monitoring changes.
+ */
+async function callDatabase() {
+  console.log('[DATABASE] start');
+  const db = firebase.database();
+  const ref = db.ref('abc/def');
+  ref.on('value', snap => {
+    if (snap.exists()) {
+      console.log(`[DATABASE] value: ${JSON.stringify(snap.val())}`);
+    } else {
+      console.log("[DATABASE] Snapshot doesn't exist");
+    }
+  });
+  console.log('[DATABASE] creating (expect to see snapshot data)');
+  await ref.set({ text: 'string 123 xyz' });
+  console.log('[DATABASE] updating (expect to see snapshot data change)');
+  await ref.update({ number: 987 });
+  console.log("[DATABASE] deleting (expect to see snapshot doesn't exist)");
+  await ref.remove();
+  ref.off();
+}
+
+/**
+ * Messaging smoke test.
+ * Call getToken(), it won't work on localhost, just a minimal test to make
```

Comment:
what wouldn't it work on localhost?

Reply:
Also added this same function to the modular sample app.